app/main.py

```diff
@@ -4,0 +1,53 @@
+from datetime import datetime
+
+from app.trip_cost import calculate_trip_cost
+from app.file_opening import file_open
+
+
+def shop_trip() -> None:
+    user_data = file_open("config.json")
+    fuel_price = user_data["fuel_price"]
+    customers = user_data["customers"]
+    shops = user_data["shops"]
+    for customer in customers:
+        cheeper_shop = dict()
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in shops:
+            cost = calculate_trip_cost(customer, shop, fuel_price)
+            cheeper_shop.update({cost: shop})
+            print(f"{customer.name}'s trip "
+                  f"to the {shop.name} costs "
+                  f"{cost}")
+        min_cost = min(list(cheeper_shop.keys()))
```

Comment:
Ensure that `cheeper_shop` is not empty before calling `min()`. If it's empty, `min()` will raise a `ValueError`. Consider adding a check to handle this case.